string new_state
---
bool success